src/pages/user-descriptor/line-graph.jsx

```diff
@@ -63,216 +49,225 @@ sortedTimeArray.sort(function(a, b) {
   b = b.split("-");
   return new Date(a[1], a[0], 1) - new Date(b[1], b[0], 1);
 });
-
-// dataArray.sort(compareValues('weight'));
-let weightArray = [];
-for (let j = 0; j < dataArray.length; j++) {
-  weightArray[j] = dataArray[j].weight;
-}
-
-// temporary dataset used to populate graph
-let rangeLength = 11;
-let rangearraym = new Array(rangeLength);
-rangearraym.fill(0);
-let rangearrayf = new Array(rangeLength);
-rangearrayf.fill(0);
-
-// finding range of users weight
-function weightRange(WTObj, weightRangeArray) {
-  let indx;
-  if (WTObj.weight >= 40 && WTObj.weight <= 50) {
-    weightRangeArray[0] = weightRangeArray[0] + 1;
-    indx = WTObj.weight;
-  } else if (WTObj.weight >= 51 && WTObj.weight <= 100) {
-    weightRangeArray[1] = weightRangeArray[1] + 1;
-    indx = WTObj.weight;
-  } else if (WTObj.weight >= 101 && WTObj.weight <= 150) {
-    weightRangeArray[2] = weightRangeArray[2] + 1;
-    indx = WTObj.weight;
-  } else if (WTObj.weight >= 151 && WTObj.weight <= 200) {
-    weightRangeArray[3] = weightRangeArray[3] + 1;
-    indx = WTObj.weight;
-  } else if (WTObj.weight >= 201 && WTObj.weight <= 300) {
-    weightRangeArray[4] = weightRangeArray[4] + 1;
-    indx = WTObj.weight;
-  } else if (WTObj.weight >= 301) {
-    weightRangeArray[5] = weightRangeArray[5] + 1;
-    indx = WTObj.weight;
-  } else {
-    indx = 0;
+export function LineGraph() {
```

Comment:
What is the reason for this to turn into a function? Is it because of the query?

Reply:
yes